### PR TITLE
Various sbt project fixes and updates

### PR DESCRIPTION
**Pull request by jonas**
- Removes unused plugins
- Fix MiMa error logging
- Bump Paradox and sbt-unidoc to latest version (see generated [unidoc](https://jonas.github.io/akka-http/unidoc) and [javaunidoc](https://jonas.github.io/akka-http/javaunidoc))
- Ignore org.specs2 from scaladoc

### CODE REVIEW

**Comment by jonas** [comment]
PLS BUILD ALL

File: project/Doc.scala [hunk 1]
```diff
@@ -139,17 +141,15 @@ object UnidocRoot extends AutoPlugin {
...
 
   override lazy val projectSettings = UnidocRoot.CliOptions.genjavadocEnabled.ifTrue(
-    genjavadocExtraSettings ++ Seq(
+    Seq(
       javacOptions in compile += "-Xdoclint:none",
       javacOptions in test += "-Xdoclint:none",
       javacOptions in doc += "-Xdoclint:none",
       scalacOptions in Compile += "-P:genjavadoc:fabricateParams=true",
-      unidocGenjavadocVersion in Global := "0.10",
-      // FIXME: see https://github.com/akka/akka-http/issues/230
```
Comment (2beaucoup) [inline project/Doc.scala]: Scould we reference #230 in commit msg or PR?

File: project/Doc.scala [hunk 2]
```diff
@@ -47,7 +49,7 @@ object Scaladoc extends AutoPlugin {
       "-groups",
       "-doc-source-url", urlString,
       "-sourcepath", base.getAbsolutePath,
-      "-skip-packages", "akka.pattern"
+      "-skip-packages", "akka.pattern:org.specs2"
```
Comment (2beaucoup) [inline project/Doc.scala]: Why is this needed?

File: project/plugins.sbt [hunk 3]
```diff
@@ -12,14 +12,13 @@ resolvers += Resolver.url("2m-sbt-plugin-releases", url("https://dl.bintray.com/
 addSbtPlugin("com.typesafe.sbt" % "sbt-multi-jvm" % "0.3.8")
 addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.1.14")
 addSbtPlugin("org.scalariform" % "sbt-scalariform" % "1.6.0")
-addSbtPlugin("com.lightbend.paradox" % "sbt-paradox" % "0.2.8")
+addSbtPlugin("com.lightbend.paradox" % "sbt-paradox" % "0.2.9")
 addSbtPlugin("com.dwijnand" % "sbt-dynver" % "1.1.1")
-addSbtPlugin("com.typesafe.sbt" % "sbt-osgi" % "0.9.1")
-addSbtPlugin("com.eed3si9n" % "sbt-unidoc" % "0.3.3")
+//addSbtPlugin("com.typesafe.sbt" % "sbt-osgi" % "0.9.1")
```
Comment (2beaucoup) [inline project/plugins.sbt]: If we agree to not support OSGI then let's just remove it.

File: project/Doc.scala (hunk 2 above)
Comment (2beaucoup) [inline project/Doc.scala]: Ouch! Could you please add a comment that includes https://issues.scala-lang.org/browse/SI-10028? Thanks!

File: project/Doc.scala (hunk 1 above)
Comment (jonas) [inline project/Doc.scala]: This doesn't fix that issue, it merely allows to remove one of the two filters that was added as a work around/

See https://github.com/akka/akka-http/blob/b825f3e9f2495d6d1562c8b504a3a07ab375782a/project/Doc.scala#L125

File: project/plugins.sbt (hunk 3 above)
Comment (jonas) [inline project/plugins.sbt]: I would be happy to, but #574 seems to still be alive. My motivation behind commenting this out is that we don't pay the cost of loading that plugin, however still leave `git grep -i osgi` helpful in case someone wants to add OGSI support.

Comment (ktoso) [inline project/plugins.sbt]: I don't think we can just ignore questions for osgi, even though I'd love to.
Don't remove the plugin yet please

File: project/Doc.scala (hunk 2 above)
Comment (jonas) [inline project/Doc.scala]: Good idea, done.

File: project/Doc.scala (hunk 1 above)
Comment (2beaucoup) [inline project/Doc.scala]: I see.

File: project/plugins.sbt (hunk 3 above)
Comment (jonas) [inline project/plugins.sbt]: OK, I've removed any surgery related to OSGI.

File: project/MiMa.scala [hunk 4]
```diff
@@ -92,9 +92,7 @@ object MiMa extends AutoPlugin {
 
     if (failures.isEmpty) mappings.map(_.right.get).toMap
     else {
-      // TODO: actually report errors using a logger, can only be done once keys have been converted to
-      // taskKeys, see https://github.com/typesafehub/migration-manager/pull/157
-      failures.flatMap(_.left.get).foreach(ex => println(ex.getMessage))
+      failures.flatMap(_.left.get).foreach(ex => logger.error(ex.getMessage))
```
Comment (jrudolph) [inline project/MiMa.scala]: Thanks, good one.

**Comment by jrudolph** [comment]
PLS BUILD

File: project/Doc.scala [hunk 5]
```diff
@@ -47,7 +49,8 @@ object Scaladoc extends AutoPlugin {
       "-groups",
       "-doc-source-url", urlString,
       "-sourcepath", base.getAbsolutePath,
-      "-skip-packages", "akka.pattern"
+      // Workaround https://issues.scala-lang.org/browse/SI-10028
+      "-skip-packages", "akka.pattern:org.specs2"
```
Comment (jrudolph) [inline project/Doc.scala]: Thanks, I guess next time it will be a completely different one but one can you do...

**Comment by jrudolph** [comment]
PLS BUILD
